### DISCUSSION TITLE
C# Socket 异步通信问题请教

### DISCUSSION THREAD

**Discussion by KiwiR1der** [BYJRK/DotNet-Discussions#36] [C# 技术交流]
初次学习 Socket 编程，没找到合适的书，参照微软学习文档写了个连接多服务端的 Demo ，想请大伙帮忙看看写的逻辑上或者异常处理有没有什么不拖的地方，写了小三天了，有点写迷糊了，想请教大佬们给些建议。

> [程序源码：SocketClientDemo](https://github.com/Boiled-Yakult/SocketClientDemo)

**Comment by dameng324** [comment]
微软提供了TcpClient和TcpListener 直接用就行，接口非常简单易用。
https://learn.microsoft.com/zh-cn/dotnet/api/system.net.sockets.tcpclient?view=net-7.0
https://learn.microsoft.com/zh-cn/dotnet/api/system.net.sockets.tcplistener?view=net-7.0

**Reply by KiwiR1der** [reply]
好嘞
